### PR TITLE
ec2: add list_keys method

### PR DESCRIPTION
**Pull request by blackboxsw**
Add ec2.list_keys operation to allow for checking if keys exist.

It allows us to look before we leap before calling either delete_key
or upload_key.

### CODE REVIEW

**Comment by paride** [comment]
@blackboxsw LGTM, but can you rebase it on master to get the CI running on it?

File: pycloudlib/ec2/cloud.py [hunk 1]
```diff
@@ -193,6 +193,12 @@ def launch(self, image_id, instance_type='t2.micro', user_data=None,
 
         return instance
 
+    def list_keys(self):
```
Comment (OddBloke) [inline pycloudlib/ec2/cloud.py:196]: Should `list_keys` this also be added to the `BaseCloud` class, as part of the standard interface that clouds are expected to implement?

Comment (OddBloke) [inline pycloudlib/ec2/cloud.py:196]: Missing a docstring here.

**Comment by paride** [comment]
Happy to see that Dan and CI agree!

File: pycloudlib/ec2/cloud.py (hunk 1 above)
Comment (blackboxsw) [reply pycloudlib/ec2/cloud.py:196]: @OddBloke it probably should, @TheRealFalcon and I were deciding intially to cowboy in implementations for the specific clouds in our first iterations and look to consolidated common methods after we have ec2, azure and oci in place. But, this is a fairly straightforward  method that I believe we need on most clouds.

Comment (blackboxsw) [reply pycloudlib/ec2/cloud.py:196]: Although, that said, lxd wouldn't need this feature. But, LXD/C will be an outlier for any of the key-related methods

Comment (blackboxsw) [reply pycloudlib/ec2/cloud.py:196]: done. pushed up a BaseCloud.list_keys. It makes sense for most clouds to support this

File: pycloudlib/cloud.py [hunk 2]
```diff
@@ -117,6 +117,14 @@ def snapshot(self, instance_id, clean=True, wait=True):
         """
         raise NotImplementedError
 
+    def list_keys(self):
+        """List ssh key names present on the cloud for accessing instances.
+
+        Returns:
+           A list of strings of key pair names accessible to the cloud.
+        """
+        return NotImplementedError
```
Comment (OddBloke) [inline pycloudlib/cloud.py:126]:
```suggestion
        raise NotImplementedError
```

**Comment by paride** [comment]
Dan's approval is required because of his "request changes" review.

**Pull request merged**